newIDE/app/src/UI/Menu/MaterialUIMenuImplementation.js

```diff
@@ -8,46 +9,113 @@ import ListItemIcon from '@material-ui/core/ListItemIcon';
 import ListItemText from '@material-ui/core/ListItemText';
 import Divider from '@material-ui/core/Divider';
 import Fade from '@material-ui/core/Fade';
+import makeStyles from '@material-ui/styles/makeStyles';
+import { adaptAcceleratorString } from '../AcceleratorString';
+import { type MenuItemTemplate } from './Menu.flow';
+
+const useStyles = makeStyles({
+  popOverRoot: {
+    // Put a `pointer-events: none` on the root of the "popover" which is showing
+    // submenus as only the menu is supposed to receive clicks.
+    pointerEvents: 'none',
+  },
+});
 
 const styles = {
-  menuItemWithSubMenu: { justifyContent: 'space-between' },
   divider: { marginLeft: 16, marginRight: 16 },
+  labelWithAccelerator: {
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  accelerator: { opacity: 0.65, marginLeft: 16 },
+  menuItemWithSubMenu: { height: 32, justifyContent: 'space-between' },
+  menuItem: {
+    // Force every menu item to have the same height, even if it's a submenu
+    // or if it has an icon.
+    height: 32,
+  },
 };
 
 const SubMenuItem = ({ item, buildFromTemplate }) => {
-  const [menuOpen, setMenuOpen] = useState(false);
-  const anchorElement = useRef(null);
-  const setAnchorElement = useCallback(element => {
-    anchorElement.current = element;
-  }, []);
+  const popoverStyles = useStyles();
+  const currentlyHovering = React.useRef(false);
+  const [anchorElement, setAnchorElement] = React.useState(null);
+
+  const handleOpen = event => {
+    // $FlowFixMe - even if not defined, not a problem.
+    if (item.enabled === false) {
+      return;
+    }
+
+    if (!anchorElement) {
+      setAnchorElement(event.currentTarget);
+    }
+  };
+
+  function handleHover() {
+    // When we hover the menu item or the submenu, we remember it
+    // so it's not closed.
+    currentlyHovering.current = true;
+  }
+
+  function handleClose() {
+    setAnchorElement(null);
+  }
+
+  function handleLeave() {
+    // Unless overwrote in the meantime, we consider that
+    // we're not hovering the menu anymore...
+    currentlyHovering.current = false;
+
+    // ...But give 75ms to the user before closing the menu,
+    // if it the menu or the item was not hovered again in the meantime.
+    setTimeout(() => {
+      if (!currentlyHovering.current) {
+        handleClose();
+      }
+    }, 75);
```

Comment:
is 75 based on a scientific reasoning? It's probably important to think it through to have a great UX. (It looks fine in your video though)

Reply:
Less than 100ms (so less than something still perceptible), more than 50ms which was really short even if you move the mouse quickly.